### DISCUSSION TITLE
Alternative Methods for Rendering SVGs and Modifying Attributes in JointJS

### DISCUSSION THREAD

**Discussion by VADDIJASWANT** [clientIO/joint#2440] [Q&A]
### Introduction

We have procured over 2000 SVG icons for our product and are attempting to render these on a JointJS paper. We referenced GitHub discussion #1964, “External SVG images,” which provided a demo on various SVG rendering methods.

While we were successful in rendering the SVGs, we encountered a significant challenge in modifying the fill and stroke colors of the SVG elements on the paper. One method from the demo involved adding a style attribute to the SVG, specifying identifiers for fill and stroke, and replacing them before encoding into URIComponent. However, our management prefers rendering SVGs directly, not in image format.

Another method involved directly using the markup, but to alter colors, we had to modify the markup and add JointJS selectors to the elements. This approach presents additional challenges, as we have more than 2000 SVGs to convert, some of which are complex with multiple linear gradient elements.

We are seeking alternative methods to render SVGs that allow for changes in fill, stroke, or other element attributes. Any guidance or suggestions would be greatly appreciated.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
It depends on how the SVG Images are built. Could you share a few examples?

1. Perhaps you could change the `fill` and `stroke` via CSS.
```css
.joint-element * { stroke: red; }
```

2. Or use images as they are, but render something above them e.g. with a [custom highlighter](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#highlighters.custom).

**Reply by VADDIJASWANT** [reply]
Please find two attached SVGs for reference.
![pump_gradient_svg](https://github.com/clientIO/joint/assets/116190897/ec6b1d9f-394e-4d5f-a793-6eb096882668)
![pump_FlatIcon](https://github.com/clientIO/joint/assets/116190897/41a4ab49-2824-4e65-9817-5b17ea411510)

**Reply by VADDIJASWANT** [reply]
Hi @kumilingus,

Could you please take a look at this issue? I'd appreciate it if you could reply when it's convenient for you.

Thank you!

**Reply by kumilingus** [reply]
In the case of an outline image, this is easy. A simple CSS changes its color:

```css
svg {
    fill: red;
}
```

I think in the case of an image with gradients, you can't do anything but manually convert them one by one.
